internal/db/schema/migrations/oss/postgres/20/10_plugin_host_secrets_hmac.up.sql

```diff
@@ -0,0 +1,33 @@
+begin;
+
+alter table host_plugin_catalog
+  add column secrets_hmac text 
```

Comment:
Instead of text, this should probably be bytes since the purpose of this field is only to return it and we can't do any other string related operations on it that would be meaningful.  I'm channeling my inner @mgaffney here :).

Reply:
I'm not sure this is an arg for storing it as bytes over text.  If we're only returning a base64 encoded format and that's all we're ever going to do, why store any other format over that?

Reply:
A couple of reasons off the top of my head:

* Text fields in the database are UTF-8 but [base64](https://en.wikipedia.org/wiki/Base64) is limited to a subset of ASCII characters.
* Any normal text transformations to the data will cause an error when the text is decoded. For example, returning the first 10-20 characters followed by an ellipse to represent a clipping, would be meaningless and cause errors.
* Base64 is for encoding binary data into text. The output of the encoding is not meant to be human readable.

Reply:
* ??? you're first point would seem to suggest that there's some problem storing a subset of ASCII characters as UTF-8.  If there is some holistic problem with storing a char like `A` in utf-8 then we've have lots of other issues.
* Any clipping of hmac data is meaningless and will cause errors. 
* All binary data that's not encoded, will be encoded when it's returned json since by definition that's what it does.

Reply:
>Any normal text transformations to the data will cause an error when the text is decoded.
The text is never decoded. It is only ever used as a string in string comparisons by e.g. Terraform.

Reply:
After discussing with @jimlambrt, we recommend the database column type be changed to `bytea`.